### PR TITLE
Update of Swedish translation

### LINKED ISSUE #914: Update Swedish translation (Engine)
The addition of new test cases and updates of some messages require that the translation needs to be updated. See [Instructions for translators] and follow the [translation steps] in the document.

[Instructions for translators]: https://github.com/zonemaster/zonemaster-engine/blob/develop/docs/Translation-translators.md
[translation steps]: https://github.com/zonemaster/zonemaster-engine/blob/develop/docs/Translation-translators.md#translation-steps

### PR DESCRIPTION
**Pull request by matsduf**
Resolves #914.

### CODE REVIEW

File: share/sv.po [hunk 1]
```diff
@@ -79,67 +79,111 @@ msgstr "TEST_CASE_END {testcase}."
 msgid "TEST_CASE_START {testcase}."
 msgstr "TEST_CASE_START {testcase}."
 
-#. BASIC:DOMAIN_NAME_LABEL_TOO_LONG
+#. BASIC:A_QUERY_NO_RESPONSES
+msgid "Nameservers did not respond to A query."
+msgstr "Namnservern svarade inte på en A-fråga."
```
Comment (mattias-p) [inline share/sv.po:84]: This msgid speaks of one A query, but there are other msgids that speak of all NS queries. I'm not sure if that inconsistency is intentional or not.

File: share/sv.po [hunk 2]
```diff
@@ -79,67 +79,111 @@ msgstr "TEST_CASE_END {testcase}."
...
-"Domändelen '{dlabel}' i domännamnet {domain} är för lång ({dlength}). "
-"Maximal längd är {max}."
+"Namnserver {ns} svarar på en SOA-fråga utan SOA-poster i svarssektionen."
 
-#. BASIC:DOMAIN_NAME_ZERO_LENGTH_LABEL
+#. BASIC:B04_NO_RESPONSE
 #, perl-brace-format
-msgid "Domain name ({domain}) has a zero-length label."
-msgstr "Domännamnet '{domain}' har en domändel med längden noll."
+msgid "Nameserver {ns} does not respond over neither UDP nor TCP."
```
Comment (mattias-p) [inline share/sv.po:103]: In the msgid I believe "responds to neither UDP nor TCP" would be better, grammatically.

File: share/sv.po [hunk 3]
```diff
@@ -79,67 +79,111 @@ msgstr "TEST_CASE_END {testcase}."
...
+#, perl-brace-format
+msgid "Nameserver {ns} does not respond to SOA queries."
+msgstr "Namnserver {ns} svarar inte på SOA-frågor."
 
-#. BASIC:HAS_PARENT
+#. BASIC:B04_NS_RECORD_NOT_AA
 #, perl-brace-format
-msgid "Parent domain '{pname}' was found for the tested domain."
-msgstr "Moderdomänen '{pname}' till den testade domänen har identifierats."
+msgid "Nameserver {ns} does not give an authoritative response on an NS query."
```
Comment (mattias-p) [inline share/sv.po:118]: In the msgid the "on" should be replaced with a "to".
There are more of these.

File: share/sv.po [hunk 4]
```diff
@@ -79,67 +79,111 @@ msgstr "TEST_CASE_END {testcase}."
...
+msgstr "Namnserver {ns} svarar inte auktoritativt på en NS-fråga."
```
Comment (mattias-p) [inline share/sv.po:119]: I believe this should be either "Namnserver {ns} svarade inte auktoritativt på en NS-fråga." or "Namnserver {ns} svarar inte auktoritativt på NS-frågor.".
There are more of these.

File: share/sv.po [hunk 5]
```diff
@@ -79,67 +79,111 @@ msgstr "TEST_CASE_END {testcase}."
...
+#. ZONE:WRONG_SOA
 #, perl-brace-format
-msgid "Nameserver {ns} did not respond to NS query."
-msgstr "Namnserver {ns} svarade inte på frågan om 'NS'."
+msgid ""
+"Nameserver {ns} responds with a wrong owner name ({owner} instead of {name}) "
+"on SOA queries."
+msgstr ""
+"Nameserver {ns} ger fel 'owner name' i svaret ({owner} istället för {name}) "
+"på SOA-fråga."
```
Comment (mattias-p) [inline share/sv.po:163]: Here it's "på SOA-fråga" but in the msgstr just above it's "på **en** NS-fråga". IMHO it should be consistent.

File: share/sv.po [hunk 6]
```diff
@@ -79,67 +79,111 @@ msgstr "TEST_CASE_END {testcase}."
...
 
-#. BASIC:A_QUERY_NO_RESPONSES
-msgid "Nameservers did not respond to A query."
-msgstr "Namnservern svarade inte på en A-fråga."
+#. BASIC:DOMAIN_NAME_LABEL_TOO_LONG
+#, perl-brace-format
+msgid ""
+"Domain name ({domain}) has a label ({dlabel}) too long ({dlength}/{max})."
```
Comment (mattias-p) [inline share/sv.po:168]: I don't remember seeing the (ACTUAL/MAXIMUM) notation anywhere else before this PR. Do you know if it's established in any way?

File: share/sv.po [hunk 7]
```diff
@@ -79,67 +79,111 @@ msgstr "TEST_CASE_END {testcase}."
...
+msgstr ""
+"Domändelen '{dlabel}' i domännamnet {domain} är för lång ({dlength}). "
+"Maximal längd är {max}."
+
+#. BASIC:DOMAIN_NAME_TOO_LONG
+#, perl-brace-format
+msgid "Domain name is too long ({fqdnlength}/{max})."
+msgstr "Domännamnet är för långt ({fqdnlength}. Maximal längd är {max}."
```
Comment (mattias-p) [inline share/sv.po:176]: Missing closing parenthesis.

File: share/sv.po [hunk 8]
```diff
@@ -79,67 +79,111 @@ msgstr "TEST_CASE_END {testcase}."
...
+
+#. BASIC:DOMAIN_NAME_ZERO_LENGTH_LABEL
+#, perl-brace-format
+msgid "Domain name ({domain}) has a zero-length label."
```
Comment (mattias-p) [inline share/sv.po:180]: This is an interesting case. It does bring some merit to presenting all domain names with the trailing dot in order to disambiguate when the second-to-last label is zero length.

File: share/sv.po [hunk 9]
```diff
@@ -655,6 +734,209 @@ msgstr ""
...
+"CDNSKEY RRset finns på namnservrarna med IP-adresserna \"{ns_ip_list}\", men "
+"inget CDS RRset."
+
+#. DS15_HAS_CDS_AND_CDNSKEY
+#, perl-brace-format
+msgid ""
+"CDNSKEY and CDS RRsets are found on nameservers that resolve to IP addresses "
+"({ns_ip_list})."
+msgstr ""
+"CDNSKEY och CDS RRsets finns namnserverarna med IP-adresseserna "
```
Comment (mattias-p) [inline share/sv.po:752]: `finns` -> `finns på`

There's an interesting mix of `namnservrarna`, `namnserverna` och `namnserverarna`. IMHO it should be consistent.

File: share/sv.po [hunk 10]
```diff
@@ -655,6 +734,209 @@ msgstr ""
...
+"from the nameservers with IP addresses \"{ns_ip_list}\"."
+msgstr ""
+"CDS-posten med \"key tag\" {keytag} stämmer inte med någon DNSKEY-post. "
+"Hämtad från namnserverna med IP-adresserna \"{ns_ip_list}\"."
+
+#. DS16_CDS_SIGNED_BY_UNKNOWN_DNSKEY
+#, perl-brace-format
+msgid ""
+"The CDS RRset is signed by RRSIG with tag {keytag} but that is not in the "
+"DNSKEY RRset. Fetched from the nameservers with P addresses \"{ns_ip_list}\"."
```
Comment (mattias-p) [inline share/sv.po:809]: There seems to be a misspelled "IP addresses" in the msgid.
There's another one like this.

File: share/sv.po [hunk 11]
```diff
@@ -655,6 +734,209 @@ msgstr ""
...
+"\"{ns_ip_list}\"."
+msgstr ""
+"DNSKEY RRset har inte signerats av den DNSKEY-post som CDS-posten med \"key "
+"tag\" {keytag} refererar till. Hämtat från namnserverna med IP-adresserna "
+"\"{ns_ip_list}\"."
+
+#. DS16_MIXED_DELETE_CDS
+#, perl-brace-format
+msgid ""
+"The CDS RRset is a mixture between \"delete\" record and other records. "
```
Comment (mattias-p) [inline share/sv.po:856]: I believe "mixture of" would be better grammatically.

File: share/sv.po [hunk 12]
```diff
@@ -655,6 +734,209 @@ msgstr ""
...
+"Fetched from the nameservers with IP addresses \"{ns_ip_list}\"."
+msgstr ""
+"CDS RRset är en blandning av poster av typen \"delete\" och andra poster. "
```
Comment (mattias-p) [inline share/sv.po:859]: "'är" -> "har"?

**Review by mattias-p — COMMENTED** [review]
The language in this PR feels more relaxed than usual. I think I like that.

I found a couple of errors in the msgstrs.

Most comments are actually on the msgids. I do realize those comments are out of scope here but I'm not sure what to do about them. Should I make a PR from them?

**Comment by matsduf** [comment]
> Most comments are actually on the msgids. I do realize those comments are out of scope here but I'm not sure what to do about them. Should I make a PR from them?

Well, they are clearly out of scope here. A PR to correct the Perl modules would be the correct way.

File: share/sv.po [hunk 13]
```diff
@@ -79,67 +79,111 @@ msgstr "TEST_CASE_END {testcase}."
...
-#. BASIC:DOMAIN_NAME_LABEL_TOO_LONG
+#. BASIC:A_QUERY_NO_RESPONSES
+msgid "Nameservers did not respond to A query."
+msgstr "Namnservern svarade inte på en A-fråga."
+
+#. BASIC:B04_MISSING_NS_RECORD
 #, perl-brace-format
 msgid ""
-"Domain name ({domain}) has a label ({dlabel}) too long ({dlength}/{max})."
+"Nameserver {ns} reponds to a NS query with no NS records in the answer "
```
[the remark this answers is not in the document]
Comment (matsduf) [reply share/sv.po:89]: Out of scope.

File: share/sv.po (hunk 1 above)
Comment (matsduf) [reply share/sv.po:84]: Out of scope.

File: share/sv.po (hunk 2 above)
Comment (matsduf) [reply share/sv.po:103]: Out of scope.

File: share/sv.po (hunk 3 above)
Comment (matsduf) [reply share/sv.po:118]: Out of scope.

File: share/sv.po (hunk 4 above)
Comment (matsduf) [reply share/sv.po:119]: It is really about a specific query.

File: share/sv.po (hunk 8 above)
Comment (matsduf) [reply share/sv.po:180]: I do not get it. Only the last label can be empty, and is actually always empty. This case shows why the domain name should be without the trailing dot, unless the domain name is the root zone.

Out of scope.

File: share/sv.po (hunk 10 above)
Comment (matsduf) [reply share/sv.po:809]: Out of scope.

File: share/sv.po (hunk 11 above)
Comment (matsduf) [reply share/sv.po:856]: Out of scope.

File: share/sv.po (hunk 4 above)
Comment (matsduf) [reply share/sv.po:119]: I see no reason to change.

File: share/sv.po (hunk 9 above)
Comment (matsduf) [reply share/sv.po:752]: It should all be `namnservrarna`. I will fix.

File: share/sv.po (hunk 12 above)
Comment (matsduf) [reply share/sv.po:859]: Both works.

**Review by mattias-p — APPROVED** [review]

**Pull request merged**